File: CultureLinkCRM/CultureLinkCRM.Web/Modules/CrmDB/Gender/GenderRow.cs
using Serenity.ComponentModel;
using Serenity.Data;
using Serenity.Data.Mapping;
using System.ComponentModel;

namespace CultureLinkCRM.CrmDB;

[ConnectionKey("Default"), Module("CrmDB"), TableName("Gender")]
[DisplayName("Gender"), InstanceName("Gender")]
[ReadPermission("Administration:General")]
[ModifyPermission("Administration:General")]
[ServiceLookupPermission("Administration:General")]
[LookupScript]
public sealed class GenderRow : Row<GenderRow.RowFields>, IIdRow, INameRow
{
    [DisplayName("Gender Id"), PrimaryKey, Identity, IdProperty]
    public int? GenderId { get => fields.GenderId[this]; set => fields.GenderId[this] = value; }

    [DisplayName("Name"), NotNull, QuickSearch, NameProperty]
    public string Name { get => fields.Name[this]; set => fields.Name[this] = value; }

    public class RowFields : RowFieldsBase
    {
        public Int32Field GenderId;
        public StringField Name;

    }
}

File: CultureLinkCRM/CultureLinkCRM.Web/Modules/ServerTypes/CrmDB/GenderRow.ts
import { getLookup, getLookupAsync, fieldsProxy } from "@serenity-is/corelib";

export interface GenderRow {
    GenderId?: number;
    Name?: string;
}

export abstract class GenderRow {
    static readonly idProperty = 'GenderId';
    static readonly nameProperty = 'Name';
    static readonly localTextPrefix = 'CrmDB.Gender';
    static readonly lookupKey = 'CrmDB.Gender';

    /** @deprecated use getLookupAsync instead */
    static getLookup() { return getLookup<GenderRow>('CrmDB.Gender') }
    static async getLookupAsync() { return getLookupAsync<GenderRow>('CrmDB.Gender') }

    static readonly deletePermission = 'Administration:General';
    static readonly insertPermission = 'Administration:General';
    static readonly readPermission = 'Administration:General';
    static readonly updatePermission = 'Administration:General';

    static readonly Fields = fieldsProxy<GenderRow>();
}